ts/session/apis/snode_api/swarmPolling.ts

```diff
@@ -400,6 +404,44 @@ export class SwarmPolling {
         window.log.info(
           `handleConfigMessagesViaLibSession of "${allDecryptedConfigMessages.length}" messages with libsession`
         );
+
+        if (returnDisplayNameOnly) {
+          let displayName = '';
+
+          try {
+            const keypair = await UserUtils.getUserED25519KeyPairBytes();
+            if (!keypair || !keypair.privKeyBytes) {
+              throw new Error('edkeypair not found for current user');
+            }
+
+            const privateKeyEd25519 = keypair.privKeyBytes;
+
+            // we take the lastest config message to create the wrapper in memory
+            const incomingConfigMessages = allDecryptedConfigMessages.map(m => ({
+              data: m.message.data,
+              hash: m.messageHash,
+            }));
+
+            await GenericWrapperActions.init('UserConfig', privateKeyEd25519, null);
+            await GenericWrapperActions.merge('UserConfig', incomingConfigMessages);
+
+            const userInfo = await UserConfigWrapperActions.getUserInfo();
+            if (!userInfo) {
+              throw new Error('UserInfo not found');
+            }
+            displayName = userInfo.name;
+          } catch (e) {
+            window.log.warn(
+              'LibSessionUtil.initializeLibSessionUtilWrappers failed with',
+              e.message
+            );
+          } finally {
+            await GenericWrapperActions.free('UserConfig');
+          }
+
+          return displayName;
+        }
```

Comment:
You can even do
```suggestion
        if (returnDisplayNameOnly) {

          try {
            const keypair = await UserUtils.getUserED25519KeyPairBytes();
            if (!keypair || !keypair.privKeyBytes) {
              throw new Error('edkeypair not found for current user');
            }

            const privateKeyEd25519 = keypair.privKeyBytes;

            // we take the lastest config message to create the wrapper in memory
            const incomingConfigMessages = allDecryptedConfigMessages.map(m => ({
              data: m.message.data,
              hash: m.messageHash,
            }));

            await GenericWrapperActions.init('UserConfig', privateKeyEd25519, null);
            await GenericWrapperActions.merge('UserConfig', incomingConfigMessages);

            const userInfo = await UserConfigWrapperActions.getUserInfo();
            if (!userInfo) {
              throw new Error('UserInfo not found');
            }
            return userInfo.name;
          } catch (e) {
            window.log.warn(
              'LibSessionUtil.initializeLibSessionUtilWrappers failed with',
              e.message
            );
          } finally {
            await GenericWrapperActions.free('UserConfig');
          }
          return '';
        }
```

and I think the finally would still be called and the return in the try be returned :exploding_head: